src/Concerns/TransactionManager.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace Jenssegers\Mongodb\Concerns;
+
+use Closure;
+use Exception;
+use MongoDB\Driver\Session;
+
+use function MongoDB\with_transaction;
+
+trait TransactionManager
+{
+    /**
+     * Use the existing or create new session and start a transaction in session
+     *
+     * In version 4.0, MongoDB supports multi-document transactions on replica sets.
+     * In version 4.2, MongoDB introduces distributed transactions, which adds support for multi-document transactions on sharded clusters and incorporates the existing support for multi-document transactions on replica sets.
+     * To use transactions on MongoDB 4.2 deployments(replica sets and sharded clusters), clients must use MongoDB drivers updated for MongoDB 4.2.
+     *
+     * @see https://docs.mongodb.com/manual/core/transactions/
+     * @return void
+     */
+    public function beginTransaction(array $options = [])
+    {
+        $session = $this->getSession();
+
+        if (!$session) {
```

Comment:
`getSession()` returns `?Session` so this seems more correct:

```suggestion
        if ($session === null) {
```

Reply:
Done